### PR TITLE
Update EF Vignette 

### PR DESCRIPTION
**Pull request by kristinawlai**
Quick fix and update of enteric fever vignette ahead of IMD presentation. 
- remove deprecated function `check_pop_data`
- Removed duplicate antibody plot
- Upgrade `est.incidence.by` to country + age 
- Update `est.incidence.by` plot 
- Reformat all code chunks to match style guide

### CODE REVIEW

**Comment by kristinawlai** [comment]
Updated the vignette to clean it up and replace plots with those from the paper. @d-morrison I don't know if these changes warrant an item in NEWS.md so the changelog is failing.

File: vignettes/articles/enteric_fever_example.Rmd [hunk 1]
```diff
@@ -76,16 +71,24 @@ curves <-
...
+### Visualize curve parameters
 
 We can graph the decay curves with an `autoplot()` method:
 
-```{r}
-curves %>% filter(antigen_iso == "HlyE_IgA"| antigen_iso == "HlyE_IgG") %>% autoplot()
+```{r, fig.width=7, fig.height=4}
+# Import longitudinal antibody parameters from OSF
+curves <- "https://osf.io/download/rtw5k/" %>%
+  load_curve_params()
```
Comment (d-morrison) [inline vignettes/articles/enteric_fever_example.Rmd]: does this step duplicate lines 69-71?

File: vignettes/articles/enteric_fever_example.Rmd [hunk 2]
```diff
@@ -103,21 +106,12 @@ Column Name | Description
 ```{r data, message = FALSE}
 
 #Import cross-sectional data from OSF and rename required variables
-xs_data <- 
-  "https://osf.io/download//n6cp3/" %>% 
-  load_pop_data() 
+xs_data <- "https://osf.io/download//n6cp3/" %>%
+  load_pop_data()
```
Comment (d-morrison) [inline vignettes/articles/enteric_fever_example.Rmd]: let's replace `load_pop_data()` with  `readr::read_rds() %>% as_pop_data()` which is a more typical workflow (see #148)

**Review by d-morrison — CHANGES_REQUESTED** [review]
Thanks for this PR! Please see comments.

**Comment by d-morrison** [comment]
> Updated the vignette to clean it up and replace plots with those from the paper. @d-morrison I don't know if these changes warrant an item in NEWS.md so the changelog is failing.

I think it's worth adding a line to NEWS.md; some of our users might appreciate the heads-up that we've improved the vignette.

**Comment by d-morrison** [comment]
also, please clear out the warnings produced by `est.incidence.by()`:

image

(the warnings tell you how to get rid of them)

**Comment by kristinawlai** [comment]
@d-morrison the OSF dataset for curve params is actually missing all of the strata so it should be used unstratified. I added noise_strata_varnames, but not sure if it helps. Also added more explanation in the text. 

> also, please clear out the warnings produced by `est.incidence.by()`:
> 
> image
> 
> (the warnings tell you how to get rid of them)

**Comment by d-morrison** [comment]
> @d-morrison the OSF dataset for curve params is actually missing all of the strata so it should be used unstratified. I added noise_strata_varnames, but not sure if it helps. Also added more explanation in the text.
> 
> > also, please clear out the warnings produced by `est.incidence.by()`:
> > image
> > (the warnings tell you how to get rid of them)

@kristinawlai Thanks for those edits! Since there are no stratifying variables in `curves`, you should be able to use `curve_strata_varnames = NULL` to remove the corresponding error.

There's still one error left - the one about unbalanced observation counts across antigen-isotypes; we don't have an easy way to mute that one right now, so we should either explain it, or remove observations that have some isotypes missing (prior to calling `est.incidence.by()`.)

**Comment by kristinawlai** [comment]
> > @d-morrison the OSF dataset for curve params is actually missing all of the strata so it should be used unstratified. I added noise_strata_varnames, but not sure if it helps. Also added more explanation in the text.
> > > also, please clear out the warnings produced by `est.incidence.by()`:
> > > image
> > > (the warnings tell you how to get rid of them)
> 
> @kristinawlai Thanks for those edits! Since there are no stratifying variables in `curves`, you should be able to use `curve_strata_varnames = NULL` to remove the corresponding error.
> 
> There's still one error left - the one about unbalanced observation counts across antigen-isotypes; we don't have an easy way to mute that one right now, so we should either explain it, or remove observations that have some isotypes missing (prior to calling `est.incidence.by()`.)

@d-morrison I thought I would try to inspect the count of antigen isotypes by adding the n for each strata to the `summary()` in line 113 (i.e. to `ab_summary` on lines 48-63 in `summary.pop_data.R`), but was having issues, so I scratched that and just added more explanation in lines 238-244 for now. Feel free to tinker with it if you'd like, otherwise I think it can be updated later.

**Comment by d-morrison** [comment]
> > > @d-morrison the OSF dataset for curve params is actually missing all of the strata so it should be used unstratified. I added noise_strata_varnames, but not sure if it helps. Also added more explanation in the text.
> > > > also, please clear out the warnings produced by `est.incidence.by()`:
> > > > image
> > > > (the warnings tell you how to get rid of them)
> > 
> > 
> > @kristinawlai Thanks for those edits! Since there are no stratifying variables in `curves`, you should be able to use `curve_strata_varnames = NULL` to remove the corresponding error.
> > There's still one error left - the one about unbalanced observation counts across antigen-isotypes; we don't have an easy way to mute that one right now, so we should either explain it, or remove observations that have some isotypes missing (prior to calling `est.incidence.by()`.)
> 
> @d-morrison I thought I would try to inspect the count of antigen isotypes by adding the n for each strata to the `summary()` in line 113 (i.e. to `ab_summary` on lines 48-63 in `summary.pop_data.R`), but was having issues, so I scratched that and just added more explanation in lines 238-244 for now. Feel free to tinker with it if you'd like, otherwise I think it can be updated later.

Fine to postpone this for now; I'll create an Issue so we don't forget to loop back to this.

**Review by d-morrison — APPROVED** [review]
Thanks - looks great!

**Pull request merged**